src/django_components/finders.py

```diff
@@ -0,0 +1,144 @@
+import os
+from fnmatch import fnmatch
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union
+
+from django.contrib.staticfiles.apps import StaticFilesConfig
+from django.contrib.staticfiles.finders import BaseFinder
+from django.contrib.staticfiles.utils import get_files
+from django.core.checks import CheckMessage, Error, Warning
+from django.core.files.storage import FileSystemStorage
+from django.utils._os import safe_join
+
+from django_components.app_settings import app_settings
+from django_components.template_loader import get_dirs
+
+# To keep track on which directories the finder has searched the static files.
+searched_locations = []
+
+
+# Custom Finder for staticfiles that searches for all files within the directories
+# defined by `COMPONENTS.dirs`.
+#
+# This is what makes it possible to define JS and CSS files in the directories as
+# defined by `COMPONENTS.dirs`, but still use the JS / CSS files with `static()` or
+# `collectstatic` command.
+class ComponentsFileSystemFinder(BaseFinder):
+    """
+    A static files finder based on ``FileSystemFinder``.
+
+    Differences:
+    - This finder uses ``COMPONENTS.dirs`` setting to locate files instead of ``STATICFILES_DIRS``.
+    - This finder ignores Python and HTML files (``.py``, ``.pyc``, ``.html``, ``.django``)
```

Comment:
I've never seen the .django ending, is that is use somewhere?

Reply:
I've seen it used in this VSCode extension https://marketplace.visualstudio.com/items?itemName=junstyle.vscode-django-support.

They use `.django`, `.dj`, `.tpl`, but I forgot about the latter two.